files/build_templates/sonic_debian_extension.j2

```diff
@@ -443,6 +443,7 @@ echo "system-health.service" | sudo tee -a $GENERATED_SERVICE_FILE
 # Copy logrotate.d configuration files
 sudo cp -f $IMAGE_CONFIGS/logrotate/logrotate.d/* $FILESYSTEM_ROOT/etc/logrotate.d/
 sudo cp $IMAGE_CONFIGS/logrotate/rsyslog.j2 $FILESYSTEM_ROOT_USR_SHARE_SONIC_TEMPLATES/
+sudo cp $IMAGE_CONFIGS/logrotate/on_demand_archived_log_clean_up.j2 $FILESYSTEM_ROOT_USR_SHARE_SONIC_TEMPLATES/
```

Comment:
>on_demand_archived_log_clean_up [](http://example.com/codeflow?start=33&length=31)

How will this j2 file be used?

Reply:
in rsyslog.j2, 
firstaction and postrotate

Reply:
during service logrotate-config restart
https://github.com/sonic-net/sonic-buildimage/blob/master/files/image_config/logrotate/logrotate-config.service
https://github.com/sonic-net/sonic-buildimage/blob/master/files/image_config/logrotate/logrotate-config.sh

logrotate-config.sh will render rsyslog.j2(which reuses on_demand_archived_log_clean_up.j2)